### PR TITLE
CLDSRV-444/CLDSRV447: test with Id and principal item and arsenal version bump

### PR DESCRIPTION
**Pull request by benzekrimaha**
### Motivation and context

In this PR :
- New unit tests were added to test the support of the Id definition
- The principal item user arn length issue is solved to match with vault's

Tickets Links : https://scality.atlassian.net/browse/CLDSRV-447
https://scality.atlassian.net/browse/CLDSRV-444

### Add tests to cover the changes

New tests added

### CODE REVIEW

**Comment by benzekrimaha** [comment]
ping

File: tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js [hunk 1]
```diff
@@ -102,5 +119,25 @@ describe('aws-sdk test put bucket policy', () => {
...
+        });
+
+        it('should put a bucket policy on bucket since Id is a string',
+        done => {
+            const params = getPolicyParamsWithId(null, 'cd3ad3d9-2776-4ef1-a904-4c229d1642e');
+            s3.putBucketPolicy(params, err =>
+                assertError(err, null, done));
+        });
+
+        it('should allow bucket policy with pincipal arn less than 2048', done => {
```
Comment (KazToozs) [inline tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: less than 2048 characters?

**Review by KazToozs — APPROVED** [review]
LGTM otherwise

File: tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js (hunk 1 above)
Comment (benzekrimaha) [reply tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: The maximum allowed is 2048 characters per AWS standard

Comment (KazToozs) [reply tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: I mean this 
```suggestion
        it('should allow bucket policy with pincipal arn less than 2048 characters', done => {
```

Comment (ghost) [inline tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: Shall we add a unit test to ensure that too long principals ARNs are rejected? To be on the safe side

**Review by ghost — COMMENTED** [review]
LGTM, just a small comment on an additional unit test

File: tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js (hunk 1 above)
Comment (benzekrimaha) [reply tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: Fix made

Comment (benzekrimaha) [reply tests/functional/aws-node-sdk/test/bucket/putBucketPolicy.js]: Test added

**Review by ghost — APPROVED** [review]

**Comment by benzekrimaha** [comment]
pring

**Comment by benzekrimaha** [comment]
ping

**Comment by benzekrimaha** [comment]
ping

**Comment by benzekrimaha** [comment]
@bert-e create_pull_requests

**Comment by benzekrimaha** [comment]
ping

**Comment by benzekrimaha** [comment]
ping

**Pull request merged**